File: python/space-age/space_age.py
class SpaceAge:
    earth_year = 0

    def __init__(self, seconds):
        self.earth_year = float(seconds)/31557600
        pass

    def on_earth(self):
        return round(self.earth_year, 2)
        pass

    def on_mercury(self):
        return round(self.earth_year/0.2408467,2)
        pass

    def on_venus(self):
        return round(self.earth_year/0.61519726,2)
        pass

    def on_mars(self):
        return round(self.earth_year/1.8808158,2)
        pass

    def on_jupiter(self):
        return round(self.earth_year/11.862615,2)
        pass

    def on_saturn(self):
        return round(self.earth_year/29.447498,2)
        pass

    def on_uranus(self):
        return round(self.earth_year/84.016846,2)
        pass

    def on_neptune(self):
        return round(self.earth_year/164.79132,2)
        pass

File: cpp/space-age/space_age.cpp
#include "space_age.h"

namespace space_age {
double mercury = 0.2408467;
double venus = 0.61519726;
double mars = 1.8808158;
double jupiter = 11.862615;
double saturn = 29.447498;
double uranus = 84.016846;
double neptune = 164.79132;

space_age::space_age(unsigned long seconds) : secs(seconds) {
    on_mercury = func(mercury);
    on_venus = func(venus);
    on_mars = func(mars);
    on_jupiter = func(jupiter);
    on_saturn = func(saturn);
    on_uranus = func(uranus);
    on_neptune = func(neptune);
}

lamda space_age::func(double factor) {
    return [factor, this]() { return on_earth() / factor; };
}

unsigned long space_age::seconds() const {
    return secs;
}

double space_age::on_earth() const {
    return static_cast<double>(secs) / 31557600u;
}
}  // namespace space_age
